docs/inputs.md

```diff
@@ -0,0 +1,319 @@
+
+## Working with Inputs in Control Code
+
+### Input Scope
+
+Inputs are available throughout the InSpec profile DSL. You can use them anywhere.
+
+```ruby
+# some_controls.rb
+
+input('outer_input', value: 1) # here
+
+control 'control-1' do
+  input('control_dsl_input', value: 2) # here too
+  describe some_resource do
+    input('test_dsl_input', value: 3) # even here
+    it { should cmp input('expectation_dsl_input') } # and yes here too
+  end
+end
+```
+
+### Setting Inputs in Control DSL
+
+When you write `input('some_name', value: 'some_value')`, you are *setting* an input value in the DSL. Because the `value:` option is present, a new value will be set.  You may also pass any other option listed in the [input option reference](#input-options-reference).
+
+### Reading Inputs in Control DSL
+
+When you call `input('some_name')`, with or without additional options, the value of the input will be resolved and returned. Note that this may involve sourcing the value from another provider, using the value set in DSL, or overriding the value provided in the same call.
+
+```ruby
+
+# You can use the value in a Ruby variable
+some_var = input('some_input_name')
+
+# Or more directly in a resource parameter
+describe file(input('important_path')) do
+  it { should exist }
+end
+
+# Or as the resource itself (this could be a string, here)
+describe input('some_setting') do
+  it { should cmp 'correct_value' }
+end
+
+# Or as the expected value
+describe file('/etc/httpd/httpd.conf') do
+  its('owner') { should_not cmp input('webserver_user') }
+end
+```
+
+The value returned can be used anywhere a Ruby value is used.
+
+## Configuring Inputs in Profile Metadata
+
+Each Chef InSpec profile has a metadata file at the top level named `inspec.yml`. In that file, you may add a section for Inputs. You may define inputs there, clearly setting options including values, type checking, and whether the input is required.
+
+```yaml
+name: my_profile
+inputs:
+- name: webserver_user  # Name is the only required field
+- name: favorite_fruit
+  value: banana         # You can set a value; priority is 30 for metadata
+- name: meaning_of_life
+  type: Numeric
+  value: 42
+  required: true
+  priority: 70
+```
+
+All [input options](#input-options-reference) are supported in metadata files.
+
+There are two major advantages to defining inputs in profile metadata:
+ 1. The inputs and their configuration are listed explicitly in simple YAML in one place - a consumer of your profile does not need to read through the control code to find the inputs.
+ 2. You can set inputs in other profiles that you depend on using profile inheritance.
+
+### Using inputs with Profile inheritance
+
+When your profile relies on another profile using the `depends` key in the metadata file, you can set (that is, override) the value of the input in the dependent profile by including the `profile` option and naming the dependent profile.
```

Comment:
I'm going to circle back to this sentence in my next pass. It needs to be clarified. For "When your profile relies on another profile using the `depends` key in the metadata file", is your profile relying on another profile using the `depends` key in the metadata file? Or is your profile using the `depends` key in the metadata file to rely on another profile? It's a bit ambiguous when first reading it.